src/components/Executions/Tables/useNodeExecutionsTableState.ts

```diff
@@ -14,10 +14,25 @@ export function useNodeExecutionsTableState({
         [value]
     );
 
-    const [
-        selectedExecution,
-        setSelectedExecution
-    ] = useState<DetailedNodeExecution | null>(null);
+    const [selectedExecutionKey, setSelectedExecutionKey] = useState<
+        string | null
+    >(null);
+
+    const selectedExecution = useMemo(
+        () =>
+            executions.find(
+                ({ cacheKey }) => cacheKey === selectedExecutionKey
+            ) || null,
```

Comment:
If `.find` doesn't find a match it will return `undefined`. Curious to know if there is a meaningful difference between returning `undefined` vs `null`?

Reply:
Short answer: No, not really. The consuming code should handle the undefined case in the same manner as the null case.

Longer answer: It's a matter of preference.

I prefer using `null` when explicitly setting a value to the "not defined" case and letting it be `undefined` when it truly was never defined or set. There are also cases where updating a value to be `undefined` is a little difficult or error-prone (like merging objects or attempting to `delete` properties). Setting to `null` feels cleaner in these cases.

To reflect that, the type of the returned value in the state is `DetailedNodeExecution | null`. It's initialized to `null` and set to `null` when the selection is cleared. So allowing it to be undefined here would cause a type error.